libraries/Bluefruit52Lib/src/bluefruit.h

```diff
@@ -178,6 +178,15 @@ class AdafruitBluefruit
 
     BLEConnection* Connection(uint16_t conn_hdl);
 
+    /*------------------------------------------------------------------*
+     * Optional semaphore for additional event handlers for SD event.
+     * It can be used for handling non-BLE  SD events 
+     *------------------------------------------------------------------*/
+    void setMultiprotocolSemaphore(SemaphoreHandle_t* p_mprot_event_semaphore) 
```

Comment:
It is not clear to me why semaphore has to passed from application, we should create it  when enable the the stack just like the ble semaphore.

Reply:
Ok, I see no reason why that could not work. I need a day, or two until I can test that variant due to other projects, but I try to prioritize this PR.

Reply:
Oh, I forget the reason, but now I recall. @hathach: I need to be able to "take" that semaphore, which I can't do if that is a private member of the Bluefruit instance.  What do you propose? 

- Either the semaphore would have to be made public, 
- or it has to be made protected and my ANT-handler class needs to be made "friend" of Bluefruit, 
- or a getter method is needed for the memory address of the semaphore. 
Either of these solution is equivalent to the current proposal having a semaphore external to Bluefruit...

Reply:
@orrmany In this case, since bluefruit.cpp has no other usage for the semaphore. I would suggest to just have an weak function hook() for mprot. Which is defined in your own application.